hydra-cluster/src/Hydra/Cluster/Faucet.hs

```diff
@@ -48,12 +52,41 @@ seedFromFaucet ::
   Lovelace ->
   -- | Marked as fuel or normal output?
   Marked ->
+  Tracer IO NodeLog ->
   IO UTxO
-seedFromFaucet RunningNode{networkId, nodeSocket} receivingVerificationKey lovelace marked = do
+seedFromFaucet RunningNode{networkId, nodeSocket} receivingVerificationKey lovelace marked tracer = do
   (faucetVk, faucetSk) <- keysFor Faucet
-  retry isSubmitTransactionException $ submitSeedTx faucetVk faucetSk
+  retryOnExceptions $ submitSeedTx faucetVk faucetSk
   waitForPayment networkId nodeSocket lovelace receivingAddress
  where
+  delay = threadDelay 1
+
+  traceException :: Exception ex => ex -> IO ()
+  traceException ex =
+    traceWith tracer $
+      MsgNodeCmdSpec
+        ( "Expected exception raised from seedFromFaucet: " <> show ex
+        )
+
+  handleSubmitException :: IO () -> SubmitTransactionException -> IO ()
+  handleSubmitException action ex = traceException ex >> delay >> retryOnExceptions action
```

Comment:
Could: only trace the `ResourceExhausted` case. The other one is actually expected here and no need to log it IMO